modules/ingest-geoip/src/main/java/org/elasticsearch/ingest/geoip/direct/TransportGetDatabaseConfigurationAction.java

```diff
@@ -82,25 +88,30 @@ protected List<DatabaseConfigurationMetadata> createActionContext(Task task, Get
                 "wildcard only supports a single value, please use comma-separated values or a single wildcard value"
             );
         }
-
+        ProjectId projectId = projectResolver.getProjectId();
         List<DatabaseConfigurationMetadata> results = new ArrayList<>();
-        PersistentTasksCustomMetadata tasksMetadata = PersistentTasksCustomMetadata.getPersistentTasksCustomMetadata(
-            clusterService.state()
+        PersistentTasksCustomMetadata tasksMetadata = PersistentTasksCustomMetadata.get(
+            clusterService.state().metadata().getProject(projectId)
```

Comment:
I think it would be a bit cleaner if we do `ProjectMetadata project = projectResolver.getProjectMetadata(clusterService.state())`, and use that here and pass that to `getMaxmindDatabases` instead of a project ID and `ClusterService`. What do you think?